### PR TITLE
#18389 certificate -> token.

### PR DESCRIPTION
**Pull request by tikinang**

### CODE REVIEW

File: src/vpn/handler_setDns.go [hunk 1]
```diff
@@ -15,7 +19,7 @@ import (
 
 var UnknownDnsManagementErr = errors.New("unknown dns management")
 
-func (h *Handler) setDns(dnsIp net.IP, clientIp net.IP, vpnNetwork net.IPNet, dnsManagement localDnsManagement) error {
+func (h *Handler) setDns(dnsIp net.IP, clientIp net.IP, vpnNetwork net.IPNet, dnsManagement dns.LocalDnsManagement) error {
```
Comment (jelito) [inline src/vpn/handler_setDns.go:22]: don't you want to move it into dns package. like detectDns

File: src/vpn/handler_public_restartVpn.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/zerops-io/zcli/src/dns"
+	"github.com/zerops-io/zcli/src/zeropsDaemonProtocol"
+)
+
+func (h *Handler) RestartVpn(ctx context.Context, token string) (*zeropsDaemonProtocol.VpnStatus, error) {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+
+	data := h.storage.Data()
+	if !h.isVpnAlive(data.ServerIp) {
```
Comment (jelito) [inline src/vpn/handler_public_restartVpn.go:16]: there is a difference between alive vpn and "started" vpn

if I've started vpn but it isn't alive (for example server is down) I wouldn't be able to restart vpn

maybe new parametr data.vpnStarted bool would be better

and then, there will be if !data.vpnStarted  { return } 
no matter if vpn is alive or not

File: src/vpn/handler_setDns.go (hunk 1 above)
Comment (tikinang) [reply src/vpn/handler_setDns.go:22]: sure

File: src/vpn/handler_public_restartVpn.go (hunk 2 above)
Comment (tikinang) [reply src/vpn/handler_public_restartVpn.go:16]: good point, but our goal is to restart vpn only if it is active. if user started vpn year ago and vpn is not active for months now, we dont want to do anything, even though "vpnStarted" would be true.
maybe the naming is littlebit confusing. would reload or restartIfActive or something more desriptive be better?

Comment (tikinang) [reply src/vpn/handler_public_restartVpn.go:16]: or is there some goroutine that checks vpn connection and trying to repair innactive "started" vpn? in that case we would probably want to try it with the new token if it was "started". if the repair of vpn is managed by user, than do nothing if vpn is not active

Comment (tikinang) [reply src/vpn/handler_public_restartVpn.go]: after consulations with @jelito and our analytic, we choose to close vpn instead of restarting it

**Pull request merged**